src/cpp/src/sequence_group.hpp

```diff
@@ -222,6 +230,50 @@ class Sequence {
         }
     }
 
+    void append_position_ids(ov::Tensor position_ids) {
+        size_t seq_len_shape_idx = position_ids.get_shape().size() == 3 ? 2 : 1;
+        size_t position_ids_len = position_ids.get_shape()[seq_len_shape_idx];
+        if (position_ids_len == 1) {
+            m_position_ids.emplace_back(position_ids);
+            return;
+        }
+        int64_t* position_ids_data = position_ids.data<int64_t>();
+        ov::Shape position_ids_elem_shape = position_ids.get_shape();
+        position_ids_elem_shape[seq_len_shape_idx] = 1;
+
+        for (size_t idx = 0; idx < position_ids_len; idx ++) {
+
+            ov::Tensor position_ids_elem(position_ids.get_element_type(), position_ids_elem_shape);
+            ov::Coordinate begin;
+            ov::Coordinate end;
+            if (position_ids_elem.get_shape().size() == 3) {
+                begin = ov::Coordinate{0, 0, idx};
+                end = ov::Coordinate{3, 1, idx + 1};
+            }
+            else {
+                begin = ov::Coordinate{0, idx};
+                end = ov::Coordinate{1, idx + 1};
+            }
+
+            ov::Tensor src_roi(position_ids, begin, end);
+            src_roi.copy_to(position_ids_elem);
```

Comment:
Would it be possible to reuse this logic? I saw the same pattern in `src/cpp/src/continuous_batching/model_runner.hpp`

Reply:
Yes, moved creating of coordinates to a static method of `Sequence` class.